dvc/stage/__init__.py

```diff
@@ -618,8 +642,10 @@ def _checkout(out, **kwargs):
             return "failed", exc.target_infos
 
     @rwlocked(read=["deps", "outs"])
-    def status(self, check_updates=False, filter_info=None):
-        ret = []
+    def status(
+        self, check_updates: bool = False, filter_info: bool = None
+    ) -> Dict[str, List[Union[str, Dict[str, str]]]]:
```

Comment:
This is quite unreadable. A sign to slightly rework the `_status*` methods below?

Reply:
First step could be to either give a name or use a proper structure to represent what is returned `Output.status()`, e.g. if:
```python
def status(self) -> EntryStatus:
    ...
```

then:

```python
def status(
    self, check_updates: bool = False, filter_info: bool = None
) -> Dict[str, List[EntryStatus]]
```

Reply:
To be honest, I don't hate verbose types. They are code-smell, but that usually means our API design is not good.
At least this makes it easy to know bad designs. I'd prefer to invent fewer types as possible. Or, use a concrete type, eg: `namedtuple` or `dataclass` for return types.